### PR TITLE
Merge console into Toolbox app / store

### PR DESCRIPTION
**Pull request by jasonLaster**
- ConsoleApp will render in the Toolbox component
- Toolbox store will include console reducers
- gets rid of the console root reducer 
- disables history and autocomplete reducer, which we can bring back later
- disables logLimit and console grouping prefs (I think in the future we should handle this stuff as a selector)...
- passes in filtersState as an action argument
- kills the webconsole hud

### CODE REVIEW

File: src/devtools/client/webconsole/actions/filters.js [hunk 1]
```diff
@@ -14,41 +14,45 @@ const {
...
+      text,
+    });
   };
 }
 
-function filterToggle(filter) {
+export function filterToggle(filter) {
   return ({ dispatch, getState, prefsService }) => {
+    const filters = getAllFilters(getState())
+    const filtersState = { ...filters, [filter]: !filters[filter] };
```
Comment (jaril) [inline src/devtools/client/webconsole/actions/filters.js:31]: `{ ...filters, [filter]: !filters[filter] }` took a good second to process, but makes sense.

File: src/devtools/client/webconsole/actions/filters.js [hunk 2]
```diff
@@ -14,41 +14,45 @@ const {
...
-function filterTextSet(text) {
-  return {
-    type: FILTER_TEXT_SET,
-    text,
+export function filterTextSet(text) {
+  return ({ dispatch, getState }) => {
+    const filtersState = { ...getAllFilters(getState()), text }
+    return dispatch({
+      type: FILTER_TEXT_SET,
+      filtersState,
```
Comment (jaril) [inline src/devtools/client/webconsole/actions/filters.js:22]: Why add the filtersState to the payload? It doesn't look like it's being handled by the reducer (`reducers/filters.js`)

File: src/devtools/client/webconsole/actions/messages.js [hunk 3]
```diff
@@ -232,9 +236,13 @@ export function setPauseExecutionPoint(executionPoint, time) {
...
 export function messageOpen(id) {
-  return {
-    type: MESSAGE_OPEN,
-    id,
+  return ({ dispatch, getState }) => {
+    const filtersState = getAllFilters(getState());
+    return dispatch({
+      type: MESSAGE_OPEN,
+      id,
+      filtersState,
```
Comment (jaril) [inline src/devtools/client/webconsole/actions/messages.js:244]: Same thing in this file. Is the `filtersState` being handled by the reducer somewhere?

File: src/devtools/client/webconsole/actions/ui.js [hunk 4]
```diff
@@ -65,8 +59,10 @@ function autocompleteToggle() {
 
 function warningGroupsToggle() {
   return ({ dispatch, getState, prefsService }) => {
+    const filtersState = getAllFilters(getState());
     dispatch({
       type: WARNING_GROUPS_TOGGLE,
+      filtersState,
```
Comment (jaril) [inline src/devtools/client/webconsole/actions/ui.js:65]: Same here. Is `filtersState` being handled?

File: src/devtools/client/webconsole/components/App.js [hunk 5]
```diff
@@ -211,41 +210,42 @@ class App extends Component {
     const filterBar = this.renderFilterBar();
     const consoleOutput = this.renderConsoleOutput();
     const jsterm = this.renderJsTerm();
-    const eager = this.renderEagerEvaluation();
+    // const eager = this.renderEagerEvaluation();
```
Comment (jaril) [inline src/devtools/client/webconsole/components/App.js:213]: Comment.

File: src/devtools/client/webconsole/components/App.js [hunk 6]
```diff
@@ -211,41 +210,42 @@ class App extends Component {
...
 
     return this.renderRootElement([
       filterBar,
       dom.div(
         { className: "flexible-output-input", key: "in-out-container" },
         consoleOutput,
         jsterm,
-        eager
+        // eager
```
Comment (jaril) [inline src/devtools/client/webconsole/components/App.js:221]: Comment.

File: src/devtools/client/webconsole/components/App.js [hunk 7]
```diff
@@ -211,41 +210,42 @@ class App extends Component {
...
-  reverseSearchInitialValue: state.ui.reverseSearchInitialValue,
-  editorMode: state.ui.editor,
-  editorWidth: state.ui.editorWidth,
-  sidebarVisible: state.ui.sidebarVisible,
-  filterBarDisplayMode: state.ui.filterBarDisplayMode,
-  eagerEvaluationEnabled: state.prefs.eagerEvaluation,
-  autocomplete: state.prefs.autocomplete,
-});
+const mapStateToProps = state => {
+  return {
```
Comment (jaril) [inline src/devtools/client/webconsole/components/App.js:238]: Weird that we don't have selectors for these, but I suppose that's fine.

File: src/devtools/client/webconsole/components/FilterBar/ConsoleSettings.js [hunk 8]
```diff
@@ -67,16 +67,16 @@ class ConsoleSettings extends Component {
     );
 
     // Warning Groups
-    items.push(
```
Comment (jaril) [inline src/devtools/client/webconsole/components/FilterBar/ConsoleSettings.js:70]: Comment.

File: src/devtools/client/webconsole/components/Input/JSTerm.js [hunk 9]
```diff
@@ -52,30 +52,30 @@ class JSTerm extends Component {
     return {
       // Returns previous or next value from the history
       // (depending on direction argument).
-      getValueFromHistory: PropTypes.func.isRequired,
```
Comment (jaril) [inline src/devtools/client/webconsole/components/Input/JSTerm.js:55]: Guessing we're just simplifying a lot here? In which case we can probably get rid of all the comments.

File: src/devtools/client/webconsole/panel.js [hunk 10]
```diff
@@ -15,13 +18,15 @@ export class WebConsolePanel {
   }
 
   async open() {
-    this.hud = new WebConsole(this.toolbox);
```
Comment (jaril) [inline src/devtools/client/webconsole/panel.js:18]: This won't be missed :P

File: src/devtools/client/webconsole/reducers/filters.js [hunk 11]
```diff
@@ -18,7 +18,7 @@ function filters(state = FilterState(), action) {
       return FilterState();
     case constants.FILTER_TEXT_SET:
       const { text } = action;
-      return cloneState(state, { [constants.FILTERS.TEXT]: text });
+      return cloneState(state, { text });
```
Comment (jaril) [inline src/devtools/client/webconsole/reducers/filters.js:21]: Don't we still have to provide a key value for text here?

File: src/devtools/client/webconsole/store.js [hunk 12]
```diff
@@ -3,39 +3,27 @@
...
-// Reducers
-const { reducers } = require("devtools/client/webconsole/reducers/index");
-
-// Middlewares
-const { thunkWithOptions } = require("devtools/client/shared/redux/middleware/thunk-with-options");
-
-function getInitialState() {
+export function getConsoleInitialState() {
   const logLimit = 1000;
   //options.logLimit || Math.max(getIntPref("devtools.hud.loglimit"), 1);
```
Comment (jaril) [inline src/devtools/client/webconsole/store.js:13]: Comments on lines like this one we can probably get rid of.

File: src/ui/components/Timeline/index.js [hunk 13]
```diff
@@ -85,7 +82,6 @@ export class Timeline extends Component {
     this.props.updateTimelineDimensions();
 
     this.toolbox.on("message-hover", this.onConsoleMessageHover);
-    this.toolbox.getPanel("console").subscribeToStore(this.onConsoleUpdate);
```
Comment (jaril) [inline src/ui/components/Timeline/index.js:88]: Nice.

File: src/ui/components/Toolbox.js [hunk 14]
```diff
@@ -453,7 +454,9 @@ class Toolbox extends React.Component {
                     active: selectedPanel == "console" || splitConsoleOpen,
                   })}
                   id="toolbox-content-console"
-                />
+                >
+                  <WebConsoleApp />
```
Comment (jaril) [inline src/ui/components/Toolbox.js:458]: YES!

File: src/devtools/client/webconsole/components/App.js (hunk 7 above)
Comment (jasonLaster) [reply src/devtools/client/webconsole/components/App.js:238]: agreed

**Pull request merged**